### PR TITLE
Fix documentation on how to get the keycode string from a `physical_keycode`

### LINKED ISSUE #82091: Documentation on physical_keycode is incorrect
### Godot version

4.1.1.stable.official

### System information

Windows 10, Vulkan forward +

### Issue description

Current documentation:
```
● Key physical_keycode [default: 0]set_physical_keycode(value) setterget_physical_keycode() getter

Represents the physical location of a key on the 101/102-key US QWERTY keyboard, which corresponds to one of the Key constants.

To get a human-readable representation of the InputEventKey, use OS.get_keycode_string(event.keycode) where event is the InputEventKey.
```

A) The example doesn't use physical_keycode.
B) This will return the QWERTY keys, not the actual key letter the user should see if they're using a non-QWERTY layout.

The way to get the correct representation is `OS.get_keycode_string(DisplayServer.keyboard_get_keycode_from_physical(event.physical_keycode))`

I'm just opening an issue to correspond with a PR I'm writing.

### Steps to reproduce

Look at the InputEventKey documentation.

### Minimal reproduction project

N/A.

### PR DESCRIPTION
**Pull request by jitspoe**
* Fixes #82091

_Bugsquad edit:_ Linked issue.

### CODE REVIEW

File: doc/classes/InputEventKey.xml [hunk 1]
```diff
@@ -79,7 +79,7 @@
 		</member>
 		<member name="physical_keycode" type="int" setter="set_physical_keycode" getter="get_physical_keycode" enum="Key" default="0">
 			Represents the physical location of a key on the 101/102-key US QWERTY keyboard, which corresponds to one of the [enum Key] constants.
-			To get a human-readable representation of the [InputEventKey], use [code]OS.get_keycode_string(event.keycode)[/code] where [code]event[/code] is the [InputEventKey].
+			To get a human-readable representation of the [InputEventKey], use [code]OS.get_keycode_string(DisplayServer.keyboard_get_keycode_from_physical(event.physical_keycode))[/code] where [code]event[/code] is the [InputEventKey].
```
Comment (bruvzg) [inline doc/classes/InputEventKey.xml:82]: Probably should be just `[code]OS.get_keycode_string(event.physical_keycode)[/code]` since it's referring to the human-readable representation of a physical key, not a remapped key.

Comment (bruvzg) [reply doc/classes/InputEventKey.xml:82]: Or this portion should be left only for the `keycode` and removed from `physical_keycode` and `key_label` at all.

Comment (jitspoe) [reply doc/classes/InputEventKey.xml:82]: If you want to print what the key for the user is, though, that's what you need to do, which isn't shown in the InputEventKey documentation.  I imagine that's the common use case, though perhaps it should be worded differently.

For example, if you have an event on the physical location of the Q key (right under the 1), but the user has an AZERTY keyboard instead of QWERTY, you'd want to display "A" instead of "Q". Ex: "Press [Q] to use action" vs "Press [A] to use action."  If you're not using QWERTY, the keys will display incorrectly.

It definitely needs to be documented in some easily accessible way.

Comment (Mickeon) [reply doc/classes/InputEventKey.xml:82]: When it's difficult to describe what the heck is going on, over something so practically simple, that warrants adding a new helper method. At least, that's what it looks like to me every time *this* happens.

Ideally and personally, for this PR, it'd be nice to directly to directly lead the user to the methods via links, instead of using the code snippet. That way, the note is less "Just do this, trust me" and more "Here's why you're doing this" because the related methods offer more information to whoever is interested. But, it's more important to be correct, rather than being awkward to read.

Comment (jitspoe) [reply doc/classes/InputEventKey.xml:82]: Linking is not a bad idea.  Can that be done directly in the code snippit?  It might get too wordy otherwise.  I do feel like the names of the functions are somewhat self explanatory: keyboard_get_keycode_from_physical gets the keycode from the physical key and get_keycode_string gets the string from the keycode.  Only thing that's not explained, and I don't know why this is to explain it, is why the keyboard_get_keycode_from_physical is on the DisplayServer and not the OS or Input related classes. 🤔

Comment (YuriSizov) [reply doc/classes/InputEventKey.xml:82]: You can't link from a snippet. And being wordy is not a problem, better than being terse. This snipped should also be formatted properly as a code block, since it's really long.

```suggestion
			To get a human-readable representation of the [InputEventKey], use [method OS.get_keycode_string] in combination with [method DisplayServer.keyboard_get_keycode_from_physical]:
			[codeblocks]
			[gdscript]
			# event is InputEventKey.
			var keycode = DisplayServer.keyboard_get_keycode_from_physical(event.physical_keycode)
			OS.get_keycode_string(keycode)
			[/gdscript]
			[/codeblocks]
```

Plus the C# example.

**Comment by akien-mga** [comment]
Added @YuriSizov's suggestion, and further reworked it to be valid code, with a C# example. It's now how full `_input` function example as C# needs `@event` to be properly defined in that context.

**Review by YuriSizov — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by stephanbogner** [comment]
Are we sure the code is working correctly?
(If not I'd open an issue for `get_keycode_string`)

I am on German **QWERTZ** keyboard on a **Mac** where the following happens:
- Press <kbd>Z</kbd> → `Z` ✅
- Press <kbd>Q</kbd> → `Q` ✅
- Press <kbd>+</kbd> → `Plus` ✅ (even though I'd prefer `+`)
- Press <kbd>-</kbd> → `Minus` ✅ (even though I'd prefer `-`)
- Press <kbd>Ü</kbd> → `BracketLeft` ❌
- Press <kbd>Ä</kbd> → `Apostrophe` ❌
- Press <kbd>Ö</kbd> → `Semicolon` ❌

It's really weird, because it seems the Umlauts (ÄÖÜ) fall back to an US keyboard layout but everything else works as expected.

Also it doesn't put it out in the user's language but that's probably a different issue/topic (which would be important when showing it to the actual user).

**Comment by AThousandShips** [comment]
If you're using `physical_keycode` that's how it works, see the documentaiton:
> Represents the physical location of a key on the 101/102-key US QWERTY keyboard, which corresponds to one of the `Key` constants.

The instruction is for how to convert that physical keycode into a human readable string *of that keycode*, not of the layout specific keycode, for that use `keycode`

**Comment by stephanbogner** [comment]
@AThousandShips 

Ah ok, I understood it as _physical_ refers to the physical keyboard I have in front of me (my bad) ... but there still seems to be a bug somewhere.

I used the code from the [docs](https://docs.godotengine.org/en/stable/classes/class_inputeventkey.html#class-inputeventkey-property-physical-keycode):
```
func _input(event):
	if event is InputEventKey:
		var keycode = DisplayServer.keyboard_get_keycode_from_physical(event.physical_keycode)
		print(OS.get_keycode_string(keycode))
```

**It seems to mix layouts in a weird manner:**
As you can see from the video when typing the keyboard works as normal (not a different layout applied), but the output from the log is like a mix of the US and the German QWERTY layout, e.g.:

<kbd>Ü</kbd> = BracketLeft (US)
<kbd>#</kbd> = NumberSign (German)
<kbd>-</kbd> = Minus (German)
<kbd>ß</kbd> = Minus (US)
<kbd>´</kbd> = Equal (US)

Tested on a MacBook Pro with 14.6.1 (Sonoma) and Godot v4.2.2

https://github.com/user-attachments/assets/f6e4f4c6-eb2f-405e-8487-423f53702368

**Comment by AThousandShips** [comment]
Please open an issue to investigate this, could be many different issues, also please provide a clear log of the results (I'd suggest making sure it prints both the keycode and the physical keycode, both in human readable forms, to evaluate), and an MRP

**Comment by AThousandShips** [comment]
That doesn't look incorrect, remember that there aren't keycodes for symbols not on the US keyboard, they are mapped to the one most relevant, so `ß` should be correct, as it is where `-` is in the US keyboard, same for `Ä`, which is where `'` is on a US keyboard

Compare with the correct layout [here](https://en.wikipedia.org/wiki/QWERTY#/media/File:KB_United_States-NoAltGr.svg)

**Comment by AThousandShips** [comment]
Forgot about `key_label`

For more info see [here](https://docs.godotengine.org/en/stable/classes/class_inputeventkey.html#class-inputeventkey-property-key-label)

**Comment by stephanbogner** [comment]
Ahhhh! Thanks @bruvzg ❤️ That does the trick! I didn't expect you have to plug something else into `OS.get_keycode_string`

**I will post an example here for the people that also get confused:**

Press <kbd>Ä</kbd> on a QWERTZ keyboard will result in:
- `event.key_label` = `196`
- `OS.get_keycode_string(event.key_label)` = `Ä`
- `OS.get_keycode_string(event.keycode)` = `Apostrophe`
- `OS.get_keycode_string(event.physical_keycode)` = `Apostrophe`

**Comment by bruvzg** [comment]
- `physical_keycode` → position of the key, e.g., WASD for movement.
- `keycode` → basic Latin shortcut available on most layouts (but might be in different places), e.g., `Ctrl+C`/`Ctrl+V` for copy/paste.
- `key_label` → what's printed on the key.
- `unicode` → text input (with modifiers, and not necessarily associated with a real key press, since can be emitted by IME/virtual keyboard, etc.).

More examples (I should add this image and some info to the docs):
KeyboardTranslation

**Comment by Zireael07** [comment]
> More examples (I should add this image and some info to the docs):

Yes PLEASE!
